backend/private-graph/graph/resolver.go

```diff
@@ -703,6 +703,20 @@ func (r *Resolver) isAdminErrorSegmentOwner(ctx context.Context, error_segment_i
 	return segment, nil
 }
 
+func (r *Resolver) isAdminSavedSegmentOwner(ctx context.Context, segment_id int) (*model.SavedSegment, error) {
+	authSpan, ctx := util.StartSpanFromContext(ctx, "isAdminSavedSegmentOwner", util.ResourceName("resolver.internal.auth"))
+	defer authSpan.Finish()
+	segment := &model.SavedSegment{}
+	if err := r.DB.WithContext(ctx).Where(&model.SavedSegment{Model: model.Model{ID: segment_id}}).Take(&segment).Error; err != nil {
```

Comment:
should replace with `.Where("id = ?", segment_id)` - if segment_id is zero, the where clause is not applied and a segment is picked arbitrarily